### DISCUSSION TITLE
Default email for web git operations

### DISCUSSION THREAD

**Discussion by n1ckoates** [community/community#51895] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Web-based git operations default to the primary email on the GitHub account, which isn't great for all scenarios. For instance, I have a public email that I use for Git, and I have it added to GitHub. I'd rather not select it _every_ time I do a web git operation on GitHub, so I set it as my primary email. This isn't optimal, though, because now important account-related notifications are sent there instead, rather than my private email because it isn't primary. Also, merging a PR on web doesn't provide an option to choose an email at all, using the primary regardless.

Hiding my email on GitHub solves these issues, but introduces it's own problems: the email is now one I'd prefer not to use (a random GitHub email rather than my own), and it hides my email from my profile, which I _don't_ want.

It'd be great if I could choose an email for web git operations in settings, while _not_ setting it to my primary email.

Thanks!

**Comment by XuehaiPan** [comment]
+1 for this.

I have two emails linked to my GitHub account. I use the primary one to receive GitHub feeds. And I set the alternative email as commit author in my local `~/.gitconfig`.

GitHub always uses the primary email for web-based commits. This results in an author with two emails in the commit history. This relates to web editors (https://github.dev), PR suggestions (` ```suggestion ``` `), web-based merge commits, etc.
